app/vlinsert/journald/journald_test.go

```diff
@@ -0,0 +1,37 @@
+package journald
+
+import (
+	"testing"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/app/vlinsert/insertutils"
+)
+
+func TestPushJournaldOk(t *testing.T) {
+	f := func(src string, timestampsExpected []int64, resultExpected string) {
+		t.Helper()
+		tlp := &insertutils.TestLogMessageProcessor{}
+		cp := &insertutils.CommonParams{
+			TimeField: "__REALTIME_TIMESTAMP",
+			MsgField:  "MESSAGE",
+		}
+		n, err := parseJournaldRequest([]byte(src), tlp, cp)
+		if err != nil {
+			t.Fatalf("unexpected error: %s", err)
+		}
+
+		if err := tlp.Verify(n, timestampsExpected, resultExpected); err != nil {
+			t.Fatal(err)
+		}
+	}
+	// Single event
+	f("__REALTIME_TIMESTAMP=91723819283\nMESSAGE=Test message",
+		[]int64{91723819283000},
+		"{\"_msg\":\"Test message\"}",
+	)
+
+	// Multiple events
+	f("__REALTIME_TIMESTAMP=91723819283\nMESSAGE=Test message\n\n__REALTIME_TIMESTAMP=91723819284\nMESSAGE=Test message2",
+		[]int64{91723819283000, 91723819284000},
+		"{\"_msg\":\"Test message\"}\n{\"_msg\":\"Test message2\"}",
+	)
```

Comment:
Please add the following test case: 
```
_CURSOR=s=bcce4fb8ffcb40e9a6e05eee8b7831bf;i=5ef603;b=ec25d6795f0645619ddac9afdef453ee;m=545242e7049;t=50f1202
__REALTIME_TIMESTAMP=1423944916375353
__MONOTONIC_TIMESTAMP=5794517905481
_BOOT_ID=ec25d6795f0645619ddac9afdef453ee
_TRANSPORT=journal
_UID=1001
_GID=1001
_CAP_EFFECTIVE=0
_SYSTEMD_OWNER_UID=1001
_SYSTEMD_SLICE=user-1001.slice
_MACHINE_ID=5833158886a8445e801d437313d25eff
_HOSTNAME=bupkis
_AUDIT_LOGINUID=1001
_SELINUX_CONTEXT=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023
CODE_LINE=1
CODE_FUNC=<module>
SYSLOG_IDENTIFIER=python3
_COMM=python3
_EXE=/usr/bin/python3.4
_AUDIT_SESSION=35898
_SYSTEMD_CGROUP=/user.slice/user-1001.slice/session-35898.scope
_SYSTEMD_SESSION=35898
_SYSTEMD_UNIT=session-35898.scope
MESSAGE
^G^@^@^@^@^@^@^@foo
bar
CODE_FILE=<string>
_PID=16853
_CMDLINE=python3 -c from systemd import journal; journal.send("foo\nbar")
_SOURCE_REALTIME_TIMESTAMP=1423944916372858
```

Reply:
event's message is not valid, I've tested it locally

Reply:
It's valid, according to the doc it should be encoding the binary data:
```
Other journal fields are serialized in a special binary safe way: field name, followed by newline, followed by a binary 64-bit little endian size value, followed by the binary field data, followed by a newline as separator to the next field.
```

 See this doc example:

```
python3 -c 'from systemd import journal; journal.send("foo\nbar")'
journalctl -n1 -o export
```

Reply:
I've executed it locally, but had no
```
MESSAGE
^G^@^@^@^@^@^@^@foo
bar
```

Reply:
I received this instead
```
__CURSOR=s=e0afe8412a6a49d2bfcf66aa7927b588;i=1e43;b=f778b6e2f7584a77b991a2366612a7b5;m=2ebc1b234c;t=6252591539469;x=5aaf020c6557a0d4
__REALTIME_TIMESTAMP=1729693142455401
__MONOTONIC_TIMESTAMP=200724390732
__SEQNUM=7747
__SEQNUM_ID=e0afe8412a6a49d2bfcf66aa7927b588
_BOOT_ID=f778b6e2f7584a77b991a2366612a7b5
_UID=0
_GID=0
_MACHINE_ID=a4a970370c30a925df02a13c67167847
_HOSTNAME=ecd5e4555787
_RUNTIME_SCOPE=system
_TRANSPORT=journal
_CAP_EFFECTIVE=1ffffffffff
_SYSTEMD_CGROUP=/init.scope
_SYSTEMD_UNIT=init.scope
_SYSTEMD_SLICE=-.slice
MESSAGE
foo
bar
CODE_FILE=<stdin>
CODE_LINE=1
CODE_FUNC=<module>
SYSLOG_IDENTIFIER=python3
_COMM=python3
_EXE=/usr/bin/python3.12
_CMDLINE=python3
_PID=2676
_SOURCE_REALTIME_TIMESTAMP=1729693142455200
```

Reply:
parser implementation with C https://github.com/systemd/systemd/blob/main/src/shared/journal-importer.c#L369

Reply:
updated